poetry/utils/env.py

```diff
@@ -1002,7 +1003,20 @@ def build_venv(
 
         args.append(str(path))
 
-        return virtualenv.cli_run(args)
+        cli_result = virtualenv.cli_run(args)
+
+        print("path", str(path))
+        print("args", args)
+
+        # Exclude the venv folder from from macOS Time Machine backups
+        if sys.platform == "darwin":
+            xattr.setxattr(
+                str(path),
+                "com.apple.metadata:com_apple_backup_excludeItem",
+                b"bplist00_\x10\x11com.apple.backupd\x08\x00\x00\x00\x00\x00\x00\x01\x01\x00\x00\x00\x00\x00\x00\x00\x01\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x1c",
```

Comment:
Would it be less intimidating to use plistlib here? E.g.

```
            xattr.setxattr(
                str(path),
                "com.apple.metadata:com_apple_backup_excludeItem",
                plistlib.dumps('com.apple.backupd', fmt=plistlib.FMT_BINARY),
            )
```

I don't have a strong opinion on this, apart from it's clearer to anyone working on the code that it is just a weird encoding of the string `com.apple.backupd`. Otherwise given the length of the hex you might think it is setting a bunch of undocumented backup related settings.

(But leave the test alone to verify it's doing the right thing).

Reply:
I don't think it's worth adding another dependency for that. I went and added a comment explaining it instead, hopefully that'll do.

Reply:
Aaah, in that case I'll go ahead and change it